webkit-extension/MailPage.vala

```diff
@@ -34,69 +34,73 @@ public class Mail.Page : Object {
     }
 
     private bool on_page_user_message_received (WebKit.WebPage page, WebKit.UserMessage message) {
+        var js_context = page.get_main_frame ().get_js_context ();
         switch (message.name) {
             case "set-body-html":
-                unowned string body = message.parameters.get_string ();
-                try {
-                    page.get_dom_document ().get_document_element ().query_selector ("#message-body").set_inner_html (body);
-                    return true;
-                } catch (Error e) {
-                    warning ("Unable to set message body content: %s", e.message);
-                }
-
-                return false;
+                unowned string body_html = message.parameters.get_string ();
+                var body = js_context.evaluate ("document.querySelector(\"#message-body\")", -1);
```

Comment:
Single quotes and double quotes are interchangeable in JS. I'd be tempted to use single quotes thoughout this so we don't have to see escape backslashes in the embedded JS.

Reply:
true